src/w2popup.js

```diff
@@ -989,3 +989,95 @@ var w2confirm = function (msg, title, callBack) {
         }
     };
 };
+
+var w2prompt = function (options, callBack) {
+    var $ = jQuery;
+    
+    options = {
+        label       : options.label ? options.label : '',
+        value       : options.value ? options.value : '',
+        title       : options.title ? options.title : w2utils.lang('Notification'),
+        ok_text     : options.ok_text ? options.ok_text : w2utils.lang('Ok'),
+        cancel_text : options.cancel_text ? options.cancel_text : 'Cancel',
+        width       : ($('#w2ui-popup').length > 0 ? 400 : 450),
+        height      : ($('#w2ui-popup').length > 0 ? 170 : 220),
+        callBack    : callBack ? callBack : null
+    }
+    
+    if ($('#w2ui-popup').length > 0 && w2popup.status != 'closing' && w2popup.get()) {
+        if (options.width > w2popup.get().width) options.width = w2popup.get().width;
+        if (options.height > (w2popup.get().height - 50)) options.height = w2popup.get().height - 50;
+          w2popup.message({
+            width   : options.width,
+            height  : options.height,
+            body    : '<div class="w2ui-centered" style="font-size: 13px;"><label style="margin-right: 10px;">' + options.label + ':</label><input id="w2promt"></div>',
+            buttons : '<button id="Ok" class="w2ui-popup-btn w2ui-btn">' + options.ok_text + '</button><button id="Cancel" class="w2ui-popup-btn w2ui-btn">' + options.cancel_text + '</button>',
+            onOpen: function () {
+                $('#w2promt').val(options.value);
+                $('#w2promt').w2field('text');
+                $('#w2ui-popup .w2ui-message .w2ui-btn#Ok').on('click.w2confirm', function (event) {
```

Comment:
since it is not confirm, should it be "click.w2prompt", I know it does not technically matter, but it will not confuse other people thinking that it somehow related to w2cofirm